tests/test_playbooks/discovery_rule.yml

```
@@ -0,0 +1,128 @@
+          - "{{ hostgroup.ptable }}"
+        operatingsystem_state: present
+    - include_tasks: tasks/installation_medium.yml
+      vars:
+        installation_medium_name: "{{ hostgroup.os.name }} Mirror"
+        installation_medium_operatingsystems:
+          - "{{ hostgroup.os.name }} {{ hostgroup.os.major }}.{{ hostgroup.os.minor }}"
+        installation_medium_locations: "{{ hostgroup.locations }}"
+        installation_medium_organizations: "{{ hostgroup.organizations }}"
+        installation_medium_state: present
+    - include_tasks: tasks/hostgroup.yml
+      vars:
+        hostgroup_name: "Discovery hostgroup"
+        hostgroup_description: "Discovery hostgroup"
+        hostgroup_architecture: "{{ hostgroup.arch }}"
+        hostgroup_operatingsystem: "{{ hostgroup.os.name }} {{ hostgroup.os.major }}.{{ hostgroup.os.minor }}"
+        hostgroup_locations: "{{ hostgroup.locations }}"
+        hostgroup_organizations: "{{ hostgroup.organizations }}"
+        hostgroup_domain: "{{ hostgroup.domains | first }}"
+        hostgroup_subnet: "{{ hostgroup.subnet }}"
+        hostgroup_installation_medium: "{{ hostgroup.os.name }} Mirror"
+        hostgroup_ptable: "{{ hostgroup.ptable }}"
+        hostgroup_pxe_loader: "Grub2 UEFI"
+        hostgroup_parameters:
+          - name: subnet_param1
+            value: value1
+          - name: subnet_param2
+            value: value2
+        hostgroup_state: present
+    - include_tasks: tasks/discovery_rule.yml
+      vars:
+        discovery_rule_name: 'New-discovery-rule'
+        discovery_rule_state: 'absent'
+
+- hosts: tests
+  collections:
+    - theforeman.foreman
+  gather_facts: false
+  vars_files:
+    - vars/server.yml
+    - vars/hostgroup.yml
+  tasks:
+    - include_tasks: tasks/discovery_rule.yml
+      vars:
+        discovery_rule_name: 'New-discovery-rule'
+        discovery_rule_search: 'mac = aa:aa:aa:aa:aa:aa'
+        discovery_rule_hostgroup: 'Discovery hostgroup'
+        discovery_rule_hostname: 'discovered-host'
+        discovery_rule_max_count: '1'
+        discovery_rule_organizations: "{{ hostgroup.organizations }}"
+        discovery_rule_locations: "{{ hostgroup.locations }}"
+        discovery_rule_state: 'present'
+        expected_change: true
+
+    - include_tasks: tasks/discovery_rule.yml
+      vars:
+        discovery_rule_name: 'New-discovery-rule'
+        discovery_rule_search: 'mac = aa:aa:aa:aa:aa:aa'
+        discovery_rule_hostgroup: 'Discovery hostgroup'
+        discovery_rule_hostname: 'discovered-host'
+        discovery_rule_max_count: '1'
+        discovery_rule_organizations: "{{ hostgroup.organizations }}"
+        discovery_rule_locations: "{{ hostgroup.locations }}"
+        discovery_rule_state: 'present'
+        expected_change: false
+
+    - include_tasks: tasks/discovery_rule.yml
+      vars:
+        discovery_rule_name: 'New-discovery-rule'
+        discovery_rule_state: 'absent'
+        expected_change: true
```

please test for idempotency of absent, and cleanup after yourself ;-)

```suggestion
        expected_change: true

    - include_tasks: tasks/discovery_rule.yml
      vars:
        discovery_rule_name: 'New-discovery-rule'
        discovery_rule_state: 'absent'
        expected_change: false

- hosts: localhost
  collections:
    - theforeman.foreman
  gather_facts: false
  vars_files:
    - vars/server.yml
    - vars/hostgroup.yml
  tasks:
    - include_tasks: tasks/location.yml
      vars:
        location_name: "{{ item }}"
        location_state: absent
      with_items: "{{ hostgroup.locations | reverse | list }}"
    - include_tasks: tasks/organization.yml
      vars:
        organization_name: "{{ item }}"
        organization_state: absent
      with_items: "{{ hostgroup.organizations }}"
```

---

Done!